ui/src/app/cohort-search/search-group-item/search-group-item.component.tsx

```diff
@@ -0,0 +1,291 @@
+      this.setState({error: true});
+    } finally {
+      this.setState({loading: false});
+    }
+  }
+
+  get itemTitleDisplay() {
+    const {item: {type}} = this.props;
+    return `${domainToTitle(type)} ${this.pluralizedCode}`;
+  }
+
+  get pluralizedCode() {
+    const {item: {searchParameters}} = this.props;
+    return searchParameters.length > 1 ? 'Codes' : 'Code';
+  }
+
+  get selectionsDisplay() {
+    const {item: {searchParameters, type}} = this.props;
+    const formatter = (param) => {
+      let funcs = [typeDisplay, attributeDisplay];
+      if (type === DomainType.PERSON) {
+        funcs = [typeDisplay, nameDisplay, attributeDisplay];
+      } else if (type === DomainType.PHYSICALMEASUREMENT
+        || type === DomainType.VISIT
+        || type === DomainType.DRUG
+        || type === DomainType.MEASUREMENT
+        || type === DomainType.SURVEY) {
+        funcs = [nameDisplay];
+      }
+      return funcs.map(f => f(param)).join(' ').trim();
+    };
+    const sep = type === DomainType[DomainType.PERSON] ? '; ' : ', ';
+    return searchParameters.map(formatter).join(sep);
+  }
+
+  enable() {
+    triggerEvent('Enable', 'Click', 'Enable - Suppress Criteria - Cohort Builder');
+    this.setState({status: 'active'});
+    this.props.item.status = 'active';
+    this.updateSearchRequest();
+  }
+
+  suppress() {
+    triggerEvent('Suppress', 'Click', 'Snowman - Suppress Criteria - Cohort Builder');
+    this.setState({status: 'hidden'});
+    this.props.item.status = 'hidden';
+    this.updateSearchRequest();
+  }
+
+  remove() {
+    triggerEvent('Delete', 'Click', 'Snowman - Delete Criteria - Cohort Builder');
+    this.setState({status: 'pending'});
+    this.props.item.status = 'pending';
+    this.updateSearchRequest();
+    const timeout = setTimeout(() => {
+      this.updateSearchRequest(true);
+    }, 10000);
+    this.setState({timeout});
+  }
+
+  undo() {
+    triggerEvent('Undo', 'Click', 'Undo - Delete Criteria - Cohort Builder');
+    clearTimeout(this.state.timeout);
+    this.setState({status: 'active'});
+    this.props.item.status = 'active';
+    this.updateSearchRequest();
+  }
+
+  updateSearchRequest(remove?: boolean) {
+    const sr = searchRequestStore.getValue();
+    const {item, groupId, role, updateGroup} = this.props;
+    const groupIndex = sr[role].findIndex(grp => grp.id === groupId);
+    if (groupIndex > -1) {
+      const itemIndex = sr[role][groupIndex].items.findIndex(it => it.id === item.id);
+      if (itemIndex > -1) {
+        if (remove) {
+          sr[role][groupIndex].items = sr[role][groupIndex].items.filter(it => it.id !== item.id);
+          searchRequestStore.next(sr);
+        } else {
+          sr[role][groupIndex].items[itemIndex] = item;
+          searchRequestStore.next(sr);
+          updateGroup();
+        }
+      }
+    }
+  }
+
+  get typeAndStandard() {
+    const {item: {searchParameters, type}} = this.props;
+    switch (type) {
```

Comment:
We need this because we're checking existing selections when re-opening the wizard (editing an item), and currently this is the only place we do that.

Reply:
Just feels like the checks below are using static data that could be represented as a map containing key domain and matching type and standard. Then this function could use that constant map to return the proper type/standard values. But I'm fine if you like it this way. I just envision anything that is static being included in the constant.ts

Reply:
Moved to `utils.ts` as a function since we need to pass in the domain and search parameters